server/ServerFSM.cpp

```diff
@@ -565,6 +572,16 @@ namespace {
                 }
             }
 
+            if (!colour_is_new) {
```

Comment:
This block conditionally sets `colour_is_new` to false so probably you don't want the `!` here?